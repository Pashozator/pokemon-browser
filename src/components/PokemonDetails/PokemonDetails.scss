.pokemon-details {
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: row;
  display: flex;

  .pokemon-list {
    height: 100px;
  }

  .image {
    padding: 25px;

    img {
      width: 100%;
      height: 100%;
    }
  }

  .description {
    margin-left: 25px;

    h1 {
      margin-top: 0;
    }

    ul {
      margin: 0;
      padding: 0;
      list-style: none;

      li {
        margin-bottom: 15px;

        .label {
          margin-bottom: 10px;
          font-size: 12px;
          font-weight: 600;
          color: #858585;
          display: block;
          text-transform: uppercase;
        }
      }
    }
  }
}
