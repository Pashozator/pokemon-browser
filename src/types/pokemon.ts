export interface Pokemon {
	id: string;
	name: string;
	image: string;
	maxHP?: number;
	fleeRate?: number;
	maxCP?: number;
	classification?: string;
	types?: string[];
	resistant?: string[];
	weaknesses?: string[];
	evolutions?: Pokemon[];
}
