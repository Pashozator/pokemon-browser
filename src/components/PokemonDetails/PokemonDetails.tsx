import React from 'react';
import './PokemonDetails.scss';
import Card from '../Card/Card';
import { Pokemon } from '../../types/pokemon';
import PokemonEvolutionList from '../PokemonEvolutionList/PokemonEvolutionList';
import SkeletonCardLoading from '../SkeletonCardLoading/SkeletonCardLoading';
import SkeletonLoading from '../SkeletonLoading/SkeletonLoading';

interface PokemonDetailsProps {
	pokemon: Pokemon;
	loading?: boolean;
}

const PokemonDetails: React.FC<PokemonDetailsProps> = ({ pokemon, loading = false }) => {
	return (
		<div className="pokemon-details">
			{loading ? <SkeletonCardLoading /> : (
				<Card>
					<div className="image">
						<img src={pokemon?.image} alt={pokemon?.name}/>
					</div>
				</Card>
			)}
			<div className="description">
				<h1>{loading ? <SkeletonLoading style={{ width: 200 }}/> : pokemon?.name}</h1>
				<ul>
					{(loading || pokemon?.maxHP) && <li>
			  <span className="label">Max HP</span> {loading ? <SkeletonLoading style={{ width: 100 }}/> : pokemon?.maxHP}
		  </li>}
					{(loading || pokemon?.maxCP) && <li>
			  <span className="label">Max CP</span> {loading ? <SkeletonLoading style={{ width: 100 }}/> : pokemon?.maxCP}
		  </li>}
					{(loading || pokemon?.fleeRate) && <li>
			  <span className="label">Flee rate</span> {loading ? <SkeletonLoading style={{ width: 100 }}/> : pokemon?.fleeRate}
		  </li>}
					{(loading || pokemon?.classification) && <li>
			  <span className="label">Classification</span> {loading ? <SkeletonLoading style={{ width: 150 }}/> : pokemon?.classification}
		  </li>}
					{(loading || pokemon?.types) && <li>
			  <span className="label">Types</span> {loading ? <SkeletonLoading style={{ width: 100 }}/> : pokemon?.types.toString()}
		  </li>}
					{(loading || pokemon?.resistant) && <li>
			  <span className="label">Resistant</span> {loading ? <SkeletonLoading style={{ width: 150 }}/> : pokemon?.resistant.toString()}
		  </li>}
					{(loading || pokemon?.weaknesses) && <li>
			  <span className="label">Weaknesses</span> {loading ? <SkeletonLoading style={{ width: 150 }}/> : pokemon?.weaknesses.toString()}
		  </li>}
					{(loading || pokemon?.evolutions) && <li>
			  <span className="label">Evolutions</span>
						{loading ? <SkeletonLoading style={{ width: 150 }}/> : <PokemonEvolutionList evolutions={pokemon?.evolutions}/>}
		  </li>}
				</ul>
			</div>
		</div>
	);
};

export default PokemonDetails;
