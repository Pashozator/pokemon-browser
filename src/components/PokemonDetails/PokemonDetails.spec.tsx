import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PokemonDetails from './PokemonDetails';
import { Pokemon } from '../../types/pokemon';

describe('PokemonDetails component', () => {
	it('should render', () => {
		const pokemon: Pokemon = {
			id: 'test',
			name: 'Pikachu',
			image: 'image'
		};
		const wrapper = shallow(<PokemonDetails pokemon={pokemon}/>);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
