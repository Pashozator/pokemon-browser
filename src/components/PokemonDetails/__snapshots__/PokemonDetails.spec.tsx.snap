// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails component should render 1`] = `
<div
  className="pokemon-details"
>
  <Card>
    <div
      className="image"
    >
      <img
        alt="Pikachu"
        src="image"
      />
    </div>
  </Card>
  <div
    className="description"
  >
    <h1>
      Pikachu
    </h1>
    <ul />
  </div>
</div>
`;
